src/AWSS3.jl

```diff
@@ -194,27 +194,33 @@ end
 
 s3_get_meta(a...; b...) = s3_get_meta(global_aws_config(), a...; b...)
 
+function _s3_exists_file(aws::AbstractAWSConfig, bucket, path; kw...)
+    q = Dict("prefix" => path, "delimiter" => "", "max-keys" => 1)
+    l = S3.list_objects_v2(bucket, q; aws_config=aws)
+    c = get(l, "Contents", nothing)
+    isnothing(c) && return false
+    return get(c, "Key", "") == path
+end
+
+function _s3_exists_dir(aws::AbstractAWSConfig, bucket, path; kw...)
+    a = string(path)[1:end-1] * "."
+    # note that you are not allowed to use *both* `prefix` and `start-after`
+    q = Dict("delimiter" => "", "max-keys" => 1, "start-after" => a)
+    l = S3.list_objects_v2(bucket, q; aws_config=aws)
```

Comment:
will this break down if there actually is a path with a `.`? since `max-keys=1` means we might just get that one and not the next one, right?

(not that I totally get what's going on here...)

Reply:
I was careful to check that case, and as far as I can tell, no, this does not break.  I will add some unit tests.

Reply:
(And yeah, it's horrendously confusing, I'll probably have to document it better.)

Reply:
Oh, I see what you were saying... yeah, its' very tricky, so `start-after` means start *AFTER*, that means this does *NOT* pick up `dir.` the first thing it does pick up is `dir/`.  That's why this should pick up that object and only that object iff it exists.  yes, super confusing

Reply:
ohh ok, got it!